tests/unit/test_microceph.py

```diff
@@ -155,3 +156,17 @@ def test_delete_cluster_configs_with_all_configs_already_in_db(self, cclient):
         ]
         assert "rgw_keystone_url" in configs_deleted
         assert "rgw_keystone_accepted_roles" in configs_deleted
+
+    @patch("microceph._run_cmd")
+    def test_join_cluster(self, run_cmd):
+        """Test if cluster join is idempotent."""
+        run_cmd.return_value = gethostname()
```

Comment:
Nit: why not set it to a fixed string instead of gethostname()? Should not make a diff. for a test and we don't risk failing the test on some hostname resolution issue

Reply:
ack, patched gethostname.